src/pixelator/utils.py

```diff
@@ -116,6 +116,18 @@ def flatten(list_of_collections: List[Union[List, Set]]) -> List:
     return [item for sublist in list_of_collections for item in sublist]
 
 
+def batched(iterable, n):
+    """Batch data into tuples of length n. The last batch may be shorter.
+
+    Taken from python itertools recipes.
+    """
+    if n < 1:
+        raise ValueError("n must be at least one")
+    it = iter(iterable)
+    while batch := tuple(itertools.islice(it, n)):
```

Comment:
The walrus operator! I haven't found much use for it but it works very nice here.